echopop/tests/test_kriging_biomass_apportionment.py

```diff
@@ -0,0 +1,159 @@
+        # ---- Drop unaged fish 
+        # !!! TODO: pending what FEAT says, weights associated with 
+        # missing ages should be added into the 'unaged' category. 
+        # This would further mean apportioning these into `weight_strata_aged_uanged`
+        .dropna( how = 'any' )
+        # ---- Remove unsexed fish
+        .loc[ lambda df: df.sex != 'unsexed' ]
+        # ---- Bin length
+        .bin_variable( objS.biology[ 'distributions' ][ 'length' ][ 'length_interval_arr' ] , 'length' )
+        # ---- Age bin
+        .bin_variable( objS.biology[ 'distributions' ][ 'age' ][ 'age_interval_arr' ] , 'age' )
+    )
+
+    ### Sum weights within each length and age bin for each sex within each stratum
+    specimen_binned_weight = (
+        specimen_binned
+        # ---- Group weight summations across distributions of length and age 
+        # ---- for each sex within each stratum
+        .groupby( [ 'stratum_num' , 'sex' , 'length_bin' , 'age_bin' ] )
+        # ---- Sum the weights 
+        .apply( lambda df: pd.Series( { 'weight_all': df.weight.sum( ) ,
+                                        'weight_adult': df.loc[ df.age > 1 ].weight.sum( ) } ) ) 
+        # ---- Fill empty/non-existent values with 0's
+        .fillna( 0 )
+        .reset_index( )
+    )
+
+    ### Calculate the relative weight proportions of each length-age bin for each sex within each stratum
+    proportions_weight_length_age_sex = (
+        specimen_binned_weight
+        # ---- Calculate total sex-specific weights for each stratum
+        .assign( total_weight_sex_all = lambda df: df.groupby( [ 'stratum_num' , 'sex' ] )[ 'weight_all' ].transform( sum ) ,
+                 total_weight_sex_adult = lambda df: df.groupby( [ 'stratum_num' , 'sex' ] )[ 'weight_adult' ].transform( sum ) )
+        # ---- Calculate the weight proportions within each sex: from Matlab --> Len_age_key_wgt_*n
+        .assign( proportion_weight_sex_all = lambda df: df.weight_all / df.total_weight_sex_all ,
+                 proportion_weight_sex_adult = lambda df: df.weight_adult / df.total_weight_sex_adult )
+    )
+
+    ### Fill empty/non-existent values with 0's
+    proportions_weight_length_age_sex[ 'proportion_weight_sex_all' ] = (
+        proportions_weight_length_age_sex[ 'proportion_weight_sex_all' ].fillna( 0 )
+    )
+
+    proportions_weight_length_age_sex[ 'proportion_weight_sex_adult' ] = (
+        proportions_weight_length_age_sex[ 'proportion_weight_sex_adult' ].fillna( 0 )
+    )
+
+    ### Check shape 
+    test_dimensions = (
+        len( np.unique( specimen_binned.age_bin ) ) *
+        len( np.unique( specimen_binned.length_bin ) ) *
+        len( np.unique( specimen_binned.sex ) ) *
+        len( np.unique( specimen_binned.stratum_num ) )
+    )
+
+    assert obj_props_wgt_len_age_sex.shape == tuple( [ test_dimensions , 10 ] )
+
+    ### Check values
+    assert obj_props_wgt_len_age_sex.equals( proportions_weight_length_age_sex )
```

Comment:
I think it may make sense to simplify the input data a bit (e.g., reduce the number of strata to 2, reduce the number of bins to 3, etc) and actually hard-code the expected values for the output of each sections: with a reduced input dataframe, you can easily hand-calculate the values of `specimen_binned`, and it is also easy to calculate `specimen_binned_weight`, `proportions_weight_length_age_sex`, and the final output.

The reason is that here you are just repeating the code blocks you have in the function, so if you update the function, it is likely you would update these sections too, which result into the test not actually functioning.